annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/AnnotationController.java

```diff
@@ -370,7 +373,16 @@ private VariantAnnotation getVariantAnnotation(String variant)
                 variantAnnotation = variantAnnotationRepository.mapAnnotationJson(variant, annotationJSON);
 
                 // save everything to the cache as a properly parsed JSON
-                variantAnnotationRepository.saveAnnotationJson(variant, annotationJSON);
+                
+                try {
+                    variantAnnotationRepository.saveAnnotationJson(variant, annotationJSON);
+                }
+                catch (DataIntegrityViolationException e) {
```

Comment:
@angelicaochoa I would add this `catch` block for the outer `try` to avoid nested try statements... Unless we really want to catch this exception only for `saveAnnotationJson` method, and we don't want to catch it for the rest of the block.